### PR TITLE
Code Cleanup: Improve gomega matchers in several packages

### PR DESCRIPTION
**Pull request by nunnatsa**
Change all `Expect(len(var)).To(Equal(some-length))` to `Expect(var).To(HaveLen(some-length))` in pkg/virt-operator, taging/src/kubevirt.io/client-go/api and pkg/watchdog. 

If the expected length is 0, the the new matcher is `BeEmpty()`

```release-note
None
```

### CODE REVIEW

File: pkg/virt-operator/kubevirt_test.go [hunk 1]
```diff
@@ -2746,9 +2746,9 @@ var _ = Describe("KubeVirt Operator", func() {
 
 			kvTestData.controller.Execute()
 
-			Expect(len(kvTestData.controller.stores.RoleCache.List())).To(Equal(2))
-			Expect(len(kvTestData.controller.stores.RoleBindingCache.List())).To(Equal(2))
-			Expect(len(kvTestData.controller.stores.ServiceMonitorCache.List())).To(Equal(0))
+			Expect(kvTestData.controller.stores.RoleCache.List()).To(HaveLen(2))
+			Expect(kvTestData.controller.stores.RoleBindingCache.List()).To(HaveLen(2))
+			Expect(kvTestData.controller.stores.ServiceMonitorCache.List()).To(HaveLen(0))
```
Comment (EdDev) [inline pkg/virt-operator/kubevirt_test.go:2751]: You have already used `BeEmpty` as part of this commit. Is this one missed?

**Review by EdDev — CHANGES_REQUESTED** [review]
Thanks!
There is one nit inline.

It would also be nice to avoid passing 72 chars in the commit message.
Ref: https://cbea.ms/git-commit/#wrap-72

File: pkg/virt-operator/kubevirt_test.go (hunk 1 above)
Comment (nunnatsa) [reply pkg/virt-operator/kubevirt_test.go]: Thanks. Fixed.

**Review by EdDev — APPROVED** [review]
Thank you!

File: pkg/virt-operator/kubevirt_test.go [hunk 2]
```diff
@@ -2147,20 +2147,20 @@ var _ = Describe("KubeVirt Operator", func() {
...
-			Expect(len(kvTestData.controller.stores.PrometheusRuleCache.List())).To(Equal(1))
+			Expect(kvTestData.controller.stores.ServiceAccountCache.List()).To(HaveLen(3))
+			Expect(kvTestData.controller.stores.ClusterRoleCache.List()).To(HaveLen(7))
+			Expect(kvTestData.controller.stores.ClusterRoleBindingCache.List()).To(HaveLen(5))
+			Expect(kvTestData.controller.stores.RoleCache.List()).To(HaveLen(3))
+			Expect(kvTestData.controller.stores.RoleBindingCache.List()).To(HaveLen(3))
+			Expect(kvTestData.controller.stores.CrdCache.List()).To(HaveLen(12))
+			Expect(kvTestData.controller.stores.ServiceCache.List()).To(HaveLen(3))
+			Expect(kvTestData.controller.stores.DeploymentCache.List()).To(HaveLen(1))
+			Expect(kvTestData.controller.stores.DaemonSetCache.List()).To(HaveLen(0))
```
Comment (orelmisan) [inline pkg/virt-operator/kubevirt_test.go:2158]: Can `BeEmpty()` be applied here?

Comment (nunnatsa) [reply pkg/virt-operator/kubevirt_test.go]: oops. Done.

Comment (enp0s3) [reply pkg/virt-operator/kubevirt_test.go]: @nunnatsa Hi. I still see `HaveLen(0)`, have you pushed the changes?

Comment (enp0s3) [reply pkg/virt-operator/kubevirt_test.go]: @nunnatsa sorry for the noise, I didn't notice it is being changed in other commit

**Review by enp0s3 — APPROVED** [review]
/approve

File: pkg/virt-operator/kubevirt_test.go (hunk 2 above)
Comment (orelmisan) [reply pkg/virt-operator/kubevirt_test.go]: Thank you @nunnatsa

**Review by orelmisan — APPROVED** [review]

**Pull request merged**